main.rb

```diff
@@ -0,0 +1,141 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for item in self
+      yield item
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    counter = 0
+    my_each do |item|
+      yield item, counter
+      counter += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected = []
+    my_each { |item| selected.push(item) if yield(item) }
+    selected
+  end
+
+  def my_all?(to_check = nil)
+    return true if !to_check and length
+
+    my_each_with_index do |item, index|
+      next unless self[index + 1]
+
+      return false unless item == self[index + 1]
+    end
+    true
+  end
+
```

Comment:
### `my_all?` should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F).